contracts/utils/RetireToucanCarbon.sol

```diff
@@ -1038,53 +1105,39 @@ contract RetireToucanCarbon is Ownable {
 
     uint256 feeAmount;
 
+    /**
+        @notice Get KLIMA type provided to regular KLIMA, swap, then retire.
+        @param _klimaType Address of the original KLIMA token provided (Wrapped/Staked/Unstaked).
+        @param _amount Total pool tokens being retired. Expected uint with 18 decimals.
+        @param _beneficiaryAddress Address of the beneficiary if different than sender. Value is set to msg.sender if null is sent.
+        @param _beneficiaryString String that can be used to describe the beneficiary
+        @param _retirementMessage String for specific retirement message if needed.
+        @param _listTCO2 List of TCO2 token addresses to redeem pool tokens.
+        @param _toucanPool Address of pool token being used to retire.
+     */
     function retireWithKLIMA(
+        address _klimaType,
         uint256 _amount,
-        string calldata _beneficiary,
-        address _KlimaType,
+        address _beneficiaryAddress,
+        string calldata _beneficiaryString,
+        string calldata _retirementMessage,
         address[] calldata _listTCO2,
         address _toucanPool
     ) public {
         require(isToucanPool[_toucanPool], "Not a Toucan Protocol Carbon Pool");
 
         uint256 totalPoolNeeded = (_amount * (1000 + feeAmount)) / 1000;
```

Comment:
So a `fee` could be something like `0.001` ?

Reply:
Got it.
https://github.com/KlimaDAO/klimadao-solidity/blob/a7d4c5a095046966b5b4f0d4dbfa779c624f0dac/contracts/utils/RetireToucanCarbon.sol#L1496